feature-libs/product-configurator/rulebased/components/price-async/configurator-price-async.component.ts

```diff
@@ -0,0 +1,145 @@
+import { useFeatureStyles } from '@spartacus/core';
+import { ConfiguratorRouterExtractorService } from '@spartacus/product-configurator/common';
+import { Observable, filter, switchMap, tap } from 'rxjs';
+import { ConfiguratorCommonsService } from '../../core/facade/configurator-commons.service';
+import { Configurator } from '../../core/model/configurator.model';
+import { ConfiguratorPriceService } from '../price/configurator-price.component.service';
+
+const NO_PRICE: Configurator.PriceDetails = {
+  currencyIso: '',
+  formattedValue: '-',
+  value: 0,
+};
+
+export interface ConfiguratorPriceAsyncComponentOptions {
+  attributeKey: string;
+  valueName: string;
+  isLightedUp?: boolean;
+}
+
+export interface ConfiguratorValuePriceChanged {
+  source: ConfiguratorPriceAsyncComponentOptions;
+  valuePrice: Configurator.PriceDetails;
+}
+
+@Component({
+  selector: 'cx-configurator-price-async',
+  templateUrl: './configurator-price-async.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfiguratorPriceAsyncComponent {
+  @Input() options: ConfiguratorPriceAsyncComponentOptions;
+  @Output() priceChanged = new EventEmitter<ConfiguratorValuePriceChanged>();
+
+  protected configuratorCommonsService = inject(ConfiguratorCommonsService);
+  protected configRouterExtractorService = inject(
+    ConfiguratorRouterExtractorService
+  );
+  protected priceService = inject(ConfiguratorPriceService);
+
+  protected lastValuePrice = NO_PRICE;
+
+  configuration$: Observable<Configurator.Configuration> =
+    this.configRouterExtractorService.extractRouterData().pipe(
+      switchMap((routerData) => {
+        return this.configuratorCommonsService
+          .getConfiguration(routerData.owner)
+          .pipe(filter((config) => !!config.priceSupplements))
+          .pipe(
+            tap((config) => {
+              const valuePrice = this.findValuePrice(config);
+              const lastValuePrice = this.lastValuePrice;
+              if (this.valuePriceChanged(lastValuePrice, valuePrice)) {
+                this.priceChanged.emit({
+                  source: this.options,
+                  valuePrice: valuePrice,
+                });
+                this.lastValuePrice = valuePrice;
+              }
+            })
+          );
+      })
+    );
+
+  constructor() {
+    useFeatureStyles('productConfiguratorAttributeTypesV2');
+  }
+
+  protected valuePriceChanged(
+    lastValuePrice: Configurator.PriceDetails,
+    valuePrice: Configurator.PriceDetails
+  ) {
+    return (
+      lastValuePrice.value !== valuePrice.value ||
+      lastValuePrice.currencyIso !== valuePrice.currencyIso ||
+      lastValuePrice.formattedValue !== valuePrice.formattedValue
+    );
+  }
+
+  getPriceDetails(
+    configuration: Configurator.Configuration
+  ): Configurator.PriceDetails {
+    return this.findValuePrice(configuration);
+  }
+
+  protected findValuePrice(configuration: Configurator.Configuration) {
```

Comment:
Please, add a return value.
What about to rename this method to `getPriceDetails` and remove the above method.